khelo/lib/ui/flow/tournament/detail/tabs/tournament_detail_teams_tab.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:data/api/team/team_model.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:khelo/domain/extensions/context_extensions.dart';
+import 'package:khelo/domain/extensions/string_extensions.dart';
+import 'package:khelo/ui/flow/tournament/detail/tournament_detail_view_model.dart';
+import 'package:style/animations/on_tap_scale.dart';
+import 'package:style/button/bottom_sticky_overlay.dart';
+import 'package:style/button/primary_button.dart';
+import 'package:style/extensions/context_extensions.dart';
+import 'package:style/text/app_text_style.dart';
+
+import '../../../../../components/empty_screen.dart';
+import '../../../../../components/image_avatar.dart';
+import '../../../../../gen/assets.gen.dart';
+import '../../../../app_route.dart';
+
+class TournamentDetailTeamsTab extends ConsumerWidget {
+  final List<TeamModel> teams;
+  final TournamentDetailStateViewNotifier notifier;
+
+  const TournamentDetailTeamsTab({
+    super.key,
+    required this.teams,
+    required this.notifier,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    if (teams.isEmpty) {
+      return Stack(
+        children: [
+          EmptyScreen(
+            title: context.l10n.tournament_detail_teams_empty_title,
+            description: context.l10n.tournament_detail_teams_empty_description,
+            isShowButton: false,
+          ),
+          _stickyButton(context),
+        ],
+      );
+    }
+
+    return ListView.separated(
+      itemCount: teams.length,
+      padding: context.mediaQueryPadding.copyWith(top: 0) +
+          EdgeInsets.all(16).copyWith(bottom: 24),
+      itemBuilder: (context, index) {
+        return _teamCellView(context, teams[index]);
+      },
+      separatorBuilder: (context, index) => SizedBox(height: 16),
+    );
+  }
+
+  Widget _teamCellView(BuildContext context, TeamModel team) {
+    return OnTapScale(
+      onTap: () => AppRoute.teamDetail(teamId: team.id).push(context),
```

Comment:
let's check if you get any console error in performing any action in teamDetail after navigating from here.